src/js/modules/infragistics.ui.editors.js

```diff
@@ -1079,9 +1079,17 @@
 					this._updateValue(newValue);
 					this._editorInput.val(this._getDisplayValue());
 				} else {
-					this._clearValue();
-					if (this._focused !== true) {
-						this._exitEditMode();
+
+					// N.A. July 27th, 2017, #1042: Trim value, when its length is larger then the maxLenght one.
+					if (this.options.maxLength) {
+						newValue = newValue.toString().substring(0, this.options.maxLength);
+						this._updateValue(newValue);
+						this._editorInput.val(this._getDisplayValue());
```

Comment:
Having in mind that, when editor can be updated through API method - value, insert, it will also require updating the visual editor value (no matter if it is focused or not), so I suggest leaving it that way.